gcc/rust/typecheck/rust-hir-type-check-pattern.cc

```diff
@@ -53,9 +53,14 @@ TypeCheckPattern::visit (HIR::TupleStructPattern &pattern)
   TyTy::BaseType *pattern_ty = TypeCheckExpr::Resolve (&pattern.get_path ());
   if (pattern_ty->get_kind () != TyTy::TypeKind::ADT)
     {
-      rust_error_at (pattern.get_locus (),
-		     "expected tuple struct/variant, found: %s",
-		     pattern_ty->get_name ().c_str ());
+      rich_location rich_locus (line_table, pattern.get_locus ());
+      rich_locus.add_fixit_replace (
+	"a tuple struct with a similar name exists");
```

Comment:
this should only be emitted if a tuple struct with a similar name exists, otherwise that help is misleading. I would remove this for now